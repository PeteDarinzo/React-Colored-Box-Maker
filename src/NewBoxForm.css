
.NewBoxForm {
  border: solid 4px red;
  padding: 10px;
  display: inline-block;
}

.NewBoxForm-label {
  margin: 10px;
}

.NewBoxForm-input {
  margin-right: 5px;
}

.NewBoxForm-button {
  display: block;
  margin: 25px auto;
}