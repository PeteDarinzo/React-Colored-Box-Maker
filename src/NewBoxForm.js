import React, { useState } from "react";
import "./NewBoxForm.css"

const NewBoxForm = ({ addBox }) => {
  const initialState = {
    height: 50,
    width: 50,
    color: "#000000"
  }

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    addBox({ ...formData });
    setFormData(initialState); // reset form inputs to default
  }

  return (
    <form className="NewBoxForm" onSubmit={handleSubmit}>
      <label htmlFor="height" className="NewBoxForm-label">Height (pixels)</label>
      <input
        id="height"
        type="number"
        name="height"
        className="NewBoxForm-input"
        min="50"
        max="200"
        value={formData.height}
        onChange={handleChange}
      />
      <label htmlFor="width" className="NewBoxForm-label">Width (pixels)</label>
      <input
        id="width"
        type="number"
        name="width"
        className="NewBoxForm-input"
        min="50"
        max="200"
        value={formData.width}
        onChange={handleChange}
      />
      <label htmlFor="color" className="NewBoxForm-label">Color</label>
      <input
        id="color"
        type="color"
        name="color"
        className="NewBoxForm-input"
        value={formData.color}
        onChange={handleChange}
      />
      <button className="NewBoxForm-button">Add Box</button>
    </form>
  );
}

export default NewBoxForm;