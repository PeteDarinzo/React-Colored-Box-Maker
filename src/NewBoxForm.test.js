import React from "react";
import { render, fireEvent } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';



it("renders a new box form without crashing", () => {
  render(<NewBoxForm
    addBox={() => null}
  />
  );
});

it("matches the new box form snapshot", () => {
  const { asFragment } = render(<NewBoxForm
    addBox={() => null}
  />
  );
  expect(asFragment()).toMatchSnapshot();
});


