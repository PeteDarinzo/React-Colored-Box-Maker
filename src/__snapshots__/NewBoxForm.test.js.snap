// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the new box form snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      class="NewBoxForm-label"
      for="height"
    >
      Height (pixels)
    </label>
    <input
      class="NewBoxForm-input"
      id="height"
      max="200"
      min="50"
      name="height"
      type="number"
      value="50"
    />
    <label
      class="NewBoxForm-label"
      for="width"
    >
      Width (pixels)
    </label>
    <input
      class="NewBoxForm-input"
      id="width"
      max="200"
      min="50"
      name="width"
      type="number"
      value="50"
    />
    <label
      class="NewBoxForm-label"
      for="color"
    >
      Color
    </label>
    <input
      class="NewBoxForm-input"
      id="color"
      name="color"
      type="color"
      value="#000000"
    />
    <button
      class="NewBoxForm-button"
    >
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
